### PR TITLE
Created Options constructor to parse config

### PR DESCRIPTION
**Pull request by joeydlee95**

### CODE REVIEW

File: src/options.h [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 class Options {
 	public:
-		Options() {}
-		short getPort(const char* file_name);
+		Options() {};
```
Comment (Filisha) [inline src/options.h]: Do we need to have an explicit default constructor?

**Comment by evangrayk** [comment]
Gonna merge this so we can update the other PR

File: src/options.h (hunk 1 above)
Comment (joeydlee95) [reply src/options.h]: good point ill get rid of it for now

File: src/main.cc [hunk 2]
```diff
@@ -9,8 +9,12 @@ int main(int argc, char* argv[]) {
...
     }
 
-    Options opt;
-    short port_num = opt.getPort(argv[1]);
+    Options opt(argv[1]);
+    short port_num = opt.getPort();
+    if (port_num == -1) {
+        std::cerr << "Was not able to get port number from <config_file>.\n";
+        return 1;
+    }
```
Comment (evangrayk) [inline src/main.cc:17]: For future options, we might want to find a better way of signaling parsing error. Maybe we can move this error reporting into `Options` itself? Let's do that later, in any case.

File: src/options.h [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 class Options {
 	public:
-		Options() {}
-		short getPort(const char* file_name);
+		Options(const char* file_name);
+		short getPort();
+	private:
+		NginxConfig config;
```
Comment (evangrayk) [inline src/options.h:8]: minor convention: add a trailing `_` to member variables, maybe change this in the future if you get a chance haha

**Review by evangrayk — COMMENTED** [review]
Looks good! :D

**Pull request merged**